SendLog.py

```diff
@@ -682,6 +698,17 @@ def add_diagram():
     return (diagram, scripts, sessiondir)
 
 
+def add_datafile():
+    (__, __, __, __, datafiles, sessiondir, __) = init_session()
+
+    datafile = DataFile()
+    datafile.datafile_currlen = str(len(datafiles))
+    datafile.sessiondir = sessiondir
+    datafiles.append(datafile)
+
+    return (datafile, sessiondir)
```

Comment:
```suggestion
    return (datafile, sessiondir, str(len(datafiles)))
```

As mentioned, return length separately.